identity/backend/src/main/java/io/camunda/identity/usermanagement/service/UserService.java

```diff
@@ -70,23 +80,30 @@ public List<CamundaUser> findAllUsers() {
     return userRepository.loadUsers();
   }
 
-  public CamundaUser updateUser(final String username, final CamundaUserWithPassword user) {
+  @Transactional
+  public CamundaUser updateUser(final Integer id, final CamundaUserWithPassword user) {
     try {
-      if (!username.equals(user.user().username())) {
+      if (!Objects.equals(id, user.user().id())) {
+        throw new RuntimeException("user.notFound");
+      }
+      final CamundaUser existingUser = userRepository.loadUserById(id);
+      if (existingUser == null || !existingUser.username().equals(user.user().username())) {
```

Comment:
❓ I'm a little confused why we're checking the usernames here? We've already performed a check on the ID which should be sufficient

Reply:
We need to check it to be sure that the requested username has not changed, for now, this method will only throw exception in this case but later if needed we can implement the path for username change as well.